syntax = "proto3";
package Nrmk.IndyFramework;
import "device_msgs.proto";
import "common_msgs.proto";

service Device {
  rpc SetBrakes(MotorList) returns (Response) {}
  rpc SetServoAll(State) returns (Response) {}
  rpc SetServo(Servo) returns (Response) {}

  rpc SetDI(DigitalList) returns (Response) {}
  rpc SetDO(DigitalList) returns (Response) {}
  rpc SetEndDI(EndtoolSignalList) returns (Response) {}
  rpc SetEndDO(EndtoolSignalList) returns (Response) {}

  rpc SetAI(AnalogList) returns (Response) {}
  rpc SetAO(AnalogList) returns (Response) {}
  rpc SetEndAI(AnalogList) returns (Response) {}
  rpc SetEndAO(AnalogList) returns (Response) {}

  rpc ExecuteTool(Name) returns (Empty) {}

  rpc SetEndRS485Rx(EndtoolRS485Rx) returns (Response) {}
  rpc GetEndRS485Rx(Empty) returns (EndtoolRS485Rx) {}
  rpc GetEndRS485Tx(Empty) returns (EndtoolRS485Tx) {}
  rpc SetEndLedDim(EndLedDim) returns (Empty) {}

  rpc GetDI(Empty) returns (DigitalList) {}
  rpc GetDO(Empty) returns (DigitalList) {}
  rpc GetEndDI(Empty) returns (EndtoolSignalList) {}
  rpc GetEndDO(Empty) returns (EndtoolSignalList) {}

  rpc GetAI(Empty) returns (AnalogList) {}
  rpc GetAO(Empty) returns (AnalogList) {}
  rpc GetEndAI(Empty) returns (AnalogList) {}
  rpc GetEndAO(Empty) returns (AnalogList) {}

  // beckhoff_conv
  rpc GetEL5001(Empty) returns (GetEL5001Data) {}
  rpc GetEL5101(Empty) returns (GetEL5101Data) {}
  // beckhoff_conv

  rpc GetDeviceInfo(Empty) returns (DeviceInfo) {}
  rpc GetBrakeControlStyle(Empty) returns (BrakeControlStyle) {}

  /////////////// Sander ////////////////
  rpc SetSanderCommand(SanderCommand) returns (Response) {}
  rpc GetSanderCommand(Empty) returns (SanderCommand) {}

  rpc GetFTSensorData(Empty) returns (FTSensorData) {}

  /////////////// Conveyor ///////////////
  rpc GetConveyor(Empty) returns (Conveyor) {}
  rpc SetConveyorName(Name) returns (Response) {}
  rpc SetConveyorByName(Name) returns (Response) {}
  rpc SetConveyorEncoder(Encoder) returns (Response) {}
  rpc SetConveyorTrigger(Trigger) returns (Response) {}
  rpc SetConveyorOffset(Float) returns (Response) {}
  rpc SetConveyorStartingPose(PosePair) returns (Response) {}
  rpc SetConveyorTerminalPose(PosePair) returns (Response) {}
  rpc GetConveyorState(Empty) returns (ConveyorState) {}

  /////////////// Safety ////////////////
  rpc GetLoadFactors(Empty) returns (GetLoadFactorsRes) {}
  rpc SetAutoMode(SetAutoModeReq) returns (SetAutoModeRes) {}
  rpc CheckAutoMode(Empty) returns (CheckAutoModeRes) {}
  rpc CheckReducedMode(Empty) returns (CheckReducedModeRes) {}
  rpc GetSafetyFunctionState(Empty) returns (SafetyFunctionState) {}
  rpc RequestSafetyFunction(SafetyFunctionState) returns (Response) {}
  rpc GetSafetyControlData(Empty) returns (SafetyControlData) {}

  ///////////////485Gripper//////////////////////
  rpc GetGripperData(Empty) returns(GripperData) {}
  rpc SetGripperCommand(GripperCommand) returns(Empty) {}

  ///////////////Photoneo Calibration//////////////////////
  rpc AddPhotoneoCalibPoint(AddPhotoneoCalibPointReq) returns(Response){}

  ///////////////Photoneo Get Detection Retrieval//////////////////////
  rpc GetPhotoneoDetection(VisionRequest) returns(VisionResult){}
  rpc GetPhotoneoRetrieval(VisionRequest) returns(VisionResult){}
}
